src/psyclone/psyir/transformations/inline_trans.py

```diff
@@ -864,6 +892,55 @@ def validate(self, node, options=None):
                             f"with a non-unit stride: "
                             f"'{actual_arg.debug_string()}' (TODO #1646)"))
 
+        # Check for dependencies within the SymbolTable of the target
+        # routine. If any of these are used to dimension a local
+        # (automatic) array, are passed by argument and are written
+        # to before the call then we can't perform inlining.
+        for asym in routine.symbol_table.automatic_datasymbols:
+            vai = VariablesAccessInfo()
+            asym.reference_accesses(vai)
+            for sig in vai.all_signatures:
+                sym = routine_table.lookup(sig.var_name)
+                if sym not in routine_table.argument_list:
+                    # This dependency is not an argument to the routine.
+                    continue
+                actual_arg = node.arguments[
+                    routine_table.argument_list.index(sym)]
+                if not isinstance(actual_arg, Reference):
+                    # The corresponding actual argument is not a Reference
+                    # so cannot be modified prior to the call.
+                    continue
+                # What form does the dependence take?
+                for prev in actual_arg.previous_accesses():
+                    if prev is node or prev.parent is node:
+                        # Skip the Call itself and any other arguments to
+                        # the call.
+                        continue
+                    exprn = prev.ancestor(Statement, include_self=True)
+                    stmt = exprn.debug_string().strip()
+                    if isinstance(prev, (CodeBlock, Call, Kern, Loop)):
+                        raise TransformationError(
+                            f"Cannot inline routine '{routine.name}' "
+                            f"because one or more of its declarations "
+                            f"depends on '{sym.name}' which is passed by "
+                            f"argument and may be written to before the "
+                            f"call ('{stmt}').")
+                    if isinstance(prev, Reference):
+                        if prev.is_write:
+                            raise TransformationError(
+                                f"Cannot inline routine '{routine.name}' "
+                                f"because one or more of its declarations "
+                                f"depends on '{sym.name}' which is passed "
+                                f"by argument and is assigned to before "
+                                f"the call ('{stmt}').")
+                        continue
```

Comment:
Codecov complains that this is not tested

Reply:
Also seems to be OK now but will see what Codecov says.